client/windows/settings/SettingsMainWindow.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * SettingsMainContainer.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "StdInc.h"
+
+#include "SettingsMainWindow.h"
+
+#include "GeneralOptionsTab.h"
+#include "AdventureOptionsTab.h"
+#include "BattleOptionsTab.h"
+#include "OtherOptionsTab.h"
+
+#include "filesystem/ResourceID.h"
+#include "CGeneralTextHandler.h"
+#include "gui/CGuiHandler.h"
+#include "lobby/CSavingScreen.h"
+#include "widgets/Buttons.h"
+#include "widgets/Images.h"
+#include "widgets/ObjectLists.h"
+#include "CGameInfo.h"
+#include "CPlayerInterface.h"
+#include "CServerHandler.h"
+
+
+SettingsMainWindow::SettingsMainWindow(BattleInterface * parentBattleUi) : InterfaceObjectConfigurable()
+{
+	OBJ_CONSTRUCTION_CAPTURING_ALL_NO_DISPOSE;
+
+	const JsonNode config(ResourceID("config/widgets/settings/settingsMainContainer.json"));
+	addCallback("activateSettingsTab", [this](int tabId) { openTab(tabId); });
```

Comment:
Just thinking aloud, feel free to ignore:
if we were to register all callbacks for all tabs in one place (e.g. by creating SettingsMainWindow::registerCallbacks() method) we can allow moving UI elements between tabs without changes in code.